WeakAuras/SubRegionTypes/Foreground.lua

```diff
@@ -0,0 +1,42 @@
+if not WeakAuras.IsCorrectVersion() then return end
+local AddonName, Private = ...
+local L = WeakAuras.L;
+
+local function subSupports(regionType)
+    return regionType == "texture"
+    or regionType == "progresstexture"
+    or regionType == "icon"
+    or regionType == "aurabar"
+end
+
+local function noop()
+end
+
+local function SetFrameLevel(self, level)
```

Comment:
I think t the cleanest would be if the regions would have a function "SetForegroundFrameLevel", and that the foreground subregion simply calls that function.

Reply:
Tried that but couldn't make it work for Icon region, finally moved it to `Private.ApplyFrameLevel` and i think it's better